metal/drivers/sifive_buserror0.h

```diff
@@ -0,0 +1,121 @@
+/* Copyright 2020 SiFive, Inc */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#ifndef METAL__DRIVERS__SIFIVE_BUSERROR0_H
+#define METAL__DRIVERS__SIFIVE_BUSERROR0_H
+
+/*!
+ * @file sifive_buserror0.h
+ *
+ * @brief API for configuring the SiFive Bus Error Unit
+ */
+
+#include <metal/compiler.h>
+
+/*!
+ * @brief The set of possible events handled by a SiFive Bus Error Unit
+ */
+typedef enum {
+    METAL_BUSERROR_EVENT_NONE = 0,
+    __METAL_BUSERROR_EVENT_MIN = METAL_BUSERROR_EVENT_NONE,
+    METAL_BUSERROR_EVENT_INST_CORRECTABLE_ECC_ERROR = 2,
+    METAL_BUSERROR_EVENT_INST_UNCORRECTABLE_ECC_ERROR = 3,
+    METAL_BUSERRROR_EVENT_LOAD_STORE_ERROR = 5,
+    METAL_BUSERROR_EVENT_DATA_CORRECTABLE_ECC_ERROR = 6,
+    METAL_BUSERROR_EVENT_DATA_UNCORRECTABLE_ECC_ERROR = 7,
+    __METAL_BUSERROR_EVENT_MAX = METAL_BUSERROR_EVENT_DATA_UNCORRECTABLE_ECC_ERROR,
+} metal_buserror_event_t;
+
+#define METAL_BUSERROR_LOCAL_INTERRUPT_ID 128U
+
+/*!
+ * @brief The handle for a bus error unit
+ */
+struct metal_buserror {};
+
+/*!
+ * @brief Enable or disable the platform interrupt
+ *
+ * @param beu The bus error unit handle
+ * @param event The error event which would trigger the interrupt
+ * @param enabled True if the interrupt should be enabled
+ * @return 0 upon success
+ */
+int metal_buserror_set_platform_interrupt(struct metal_buserror *beu, metal_buserror_event_t event, bool enabled);
+
+/*!
+ * @brief Enable or disable the hart-local interrupt
+ *
+ * @param beu The bus error unit handle
+ * @param event The error event which would trigger the interrupt
+ * @param enabled True if the interrupt should be enabled
+ * @return 0 upon success
+ */
+int metal_buserror_set_local_interrupt(struct metal_buserror *beu, metal_buserror_event_t event, bool enabled);
```

Comment:
The intention is the handle will have the hart-id?

Reply:
Yeah, the bus error unit handle will know which hart ID it's associated with.

Reply:
Sorry, I was confused. The BEU doesn't need to know its hart ID because this is the interrupt enable within the BEU. The user knows the hart ID because the BEU handle is retrieved from an individual CPU handle.

Reply:
Plus, the BEU interrupt is permanently enabled.

Reply:
Are you planning to save the hart-id in the handle?

Reply:
It won't need it.

Reply:
That's right the beu is already specific to a hart, and there is already per hart interrupt-controller/interrupt-table. I am good. Thanks